velox/functions/lib/aggregates/SetBaseAggregate.h

```diff
@@ -50,7 +50,13 @@ class SetBaseAggregate : public exec::Aggregate {
     for (auto i = 0; i < numGroups; ++i) {
       auto* group = groups[i];
       if (isNull(group)) {
-        arrayVector->setNull(i, true);
+        if constexpr (ignoreNulls) {
+          // When ignoring nulls, if the group's accumulator is null, the
+          // corresponding result is an empty array.
+          clearNull(rawNulls, i);
```

Comment:
Should we set `rawOffsets[i] = numValues` and `rawSizes[i] = 0` here?

Reply:
I thought the result is correct as the init values for both is 0.

Reply:
Thanks for the review. Updated.